pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -703,6 +717,28 @@ func (r *RepoSyncReconciler) mapConfigMapToRepoSyncs(obj client.Object) []reconc
 	}
 
 	ctx := context.Background()
+	var sourceConfigMap corev1.ConfigMap
+	err := r.client.Get(ctx, objRef, &sourceConfigMap)
+
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// the referenced ConfigMap doesn't exist anymore, so we should delete our copy of it (if we have one).
+			newNamespace := configsync.ControllerNamespace
+			newName := core.NsReconcilerPrefix + "-" + objRef.Name + "-" + objRef.Namespace
+
+			sourceConfigMapCopy := sourceConfigMap.DeepCopy()
+			sourceConfigMapCopy.ObjectMeta = metav1.ObjectMeta{
+				Namespace: newNamespace,
+				Name:      newName,
+			}
+			if err := r.client.Delete(ctx, sourceConfigMapCopy); err != nil {
+				klog.Errorf("failed to delete copy of ConfigMap: %w", err)
+			}
+		} else {
+			klog.Errorf("failed to get referenced ConfigMap: %w", err)
+		}
+	}
+
```

Comment:
If users switch from valuesfiles to inline values, or when the list of valuesfiles change, would those previously-copied configmaps be cleaned up?

Reply:
Good catch. Added cleanup for the case of valuesFileRefs changing.